e2e/ctl_v3_user_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import "testing"
+
+func TestCtlV3UserAdd(t *testing.T)          { testCtl(t, userAddTest) }
+func TestCtlV3UserAddNoTLS(t *testing.T)     { testCtl(t, userAddTest, withCfg(configNoTLS)) }
+func TestCtlV3UserAddClientTLS(t *testing.T) { testCtl(t, userAddTest, withCfg(configClientTLS)) }
+func TestCtlV3UserAddPeerTLS(t *testing.T)   { testCtl(t, userAddTest, withCfg(configPeerTLS)) }
+func TestCtlV3UserAddTimeout(t *testing.T)   { testCtl(t, userAddTest, withDialTimeout(0)) }
+
+func TestCtlV3UserDelete(t *testing.T)          { testCtl(t, userDelTest) }
+func TestCtlV3UserDeleteNoTLS(t *testing.T)     { testCtl(t, userDelTest, withCfg(configNoTLS)) }
+func TestCtlV3UserDeleteClientTLS(t *testing.T) { testCtl(t, userDelTest, withCfg(configClientTLS)) }
+func TestCtlV3UserDeletePeerTLS(t *testing.T)   { testCtl(t, userDelTest, withCfg(configPeerTLS)) }
+func TestCtlV3UserDeleteTimeout(t *testing.T)   { testCtl(t, userDelTest, withDialTimeout(0)) }
+
+func TestCtlV3UserPasswd(t *testing.T)          { testCtl(t, userPasswdTest) }
+func TestCtlV3UserPasswdNoTLS(t *testing.T)     { testCtl(t, userPasswdTest, withCfg(configNoTLS)) }
+func TestCtlV3UserPasswdClientTLS(t *testing.T) { testCtl(t, userPasswdTest, withCfg(configClientTLS)) }
+func TestCtlV3UserPasswdPeerTLS(t *testing.T)   { testCtl(t, userPasswdTest, withCfg(configPeerTLS)) }
+func TestCtlV3UserPasswdTimeout(t *testing.T)   { testCtl(t, userPasswdTest, withDialTimeout(0)) }
+
+type userCmdDesc struct {
+	args        []string
+	expectedStr string
+	stdIn       []string
+}
+
+func userAddTest(cx ctlCtx) {
+	defer close(cx.errc)
+
+	cmdSet := map[string]userCmdDesc{
+		// Adds a user name.
+		"add1": {
+			args:        []string{"add", "username", "--interactive=false"},
```

Comment:
I noticed that for `user add` sub-command the default is true for `--interactive` flag. Hence I explicitly set it to false.